temporal-server/authorizer/token.go

```diff
@@ -0,0 +1,105 @@
+package authorizer
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type TokenInfo struct {
+	Azp           string `json:"azp"`
+	Aud           string `json:"aud"`
+	Sub           string `json:"sub"`
+	Scope         string `json:"scope"`
+	Exp           string `json:"exp"`
+	ExpiresIn     string `json:"expires_in"`
+	Email         string `json:"email"`
+	EmailVerified string `json:"email_verified"`
+	AccessType    string `json:"access_type"`
+}
+
+// TokenVerifier is an interface that defines the methods
+// to fetch token information and verify their validity.
+type TokenVerifier interface {
+	GetTokenInfo(accessToken string) (*TokenInfo, error)
+	VerifyToken(token *TokenInfo) error
+}
```

Comment:
Interfaces should usually be declared in consumer packages, not in the packages implementing them.

See: https://github.com/golang/go/wiki/CodeReviewComments#interfaces